### DISCUSSION TITLE
[YTDLError: aborted] Facing this in Heroku.

### DISCUSSION THREAD

**Discussion by SouvikUnscripted** [Hazmi35/jukebox#551] [Q&A] [answered]
Haven't changed anything, Just deployed the app through the "Deploy to Heroku" button, and it keeps telling me Error while playing music Reason: YTDLError: aborted. - How do I fix this?

**Comment by Hazmi35** [comment]
Does this happen at the start of the music?

**Comment by SouvikUnscripted** [comment]
@Hazmi35 In middle of a track. (Never happend to me in the starting of any tracks.) The error pops in and the bot leaves the voice channel.

**Comment by Hazmi35** [comment]
Is it just specific to one track? can't reproduce

**Comment by SouvikUnscripted** [comment]
Nope, I tried with some songs it happens. Is there any extra builpacks or something we need to add?

**Comment by Hazmi35** [comment]
> Nope, I tried with some songs it happens. Is there any extra builpacks or something we need to add?

No, heroku/nodejs is the only thing you need, because ffmpeg is already installed in node_modules by ffmpeg-static, what's your Heroku stack? trying to reproduce again

**Reply by Hazmi35** [reply]
Could reproduce, I'm investigating what the cause of this issue

**Comment by Hazmi35** [comment, marked as answer]
Could reproduce, this is an Issue with ytdl-core x Discord.js on Node v16 (16.5.0), creating an issue #552

**Reply by Hazmi35** [reply]
Fixed in #553, released as [Version 4.39.2](https://github.com/Hazmi35/jukebox/releases/tag/4.39.2).

This issue is actually caused by Node.js v16 changes for Stream API, this is fixed by modifying highWatermark in the ytdl-core option. 

*Why does Heroku use Node.js v16?*
Because the semver requirement for Node.js in Jukebox is ">=12.9.0" which means version 12.9.0 or newer, and Heroku chooses the latest Node.js available (16.5.0 as of the writing of this comment).
This is also fixed in the new release, **now "Deploy to Heroku" button will use "container" as a stack, and make use of Jukebox's Dockerfile, this would improve stability**
